src/components/tl/mlx5/mcast/tl_mlx5_mcast_one_sided_reliability.c

```diff
@@ -0,0 +1,228 @@
+/**
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_mlx5_mcast_one_sided_reliability.h"
+
+static ucc_status_t ucc_tl_mlx5_mcast_one_sided_setup_reliability_buffers(ucc_base_team_t *team)
+{
+    ucc_tl_mlx5_team_t            *tl_team = ucc_derived_of(team, ucc_tl_mlx5_team_t);
+    ucc_status_t                   status  = UCC_OK;
+    ucc_tl_mlx5_mcast_coll_comm_t *comm    = tl_team->mcast->mcast_comm;
+    int                            one_sided_total_slots_size, i;
+
+    /* this array keeps track of the number of recv packets from each process
+     * used in all the protocols */
+    comm->one_sided.recvd_pkts_tracker = ucc_calloc(1, comm->commsize * sizeof(uint32_t),
+                                                    "one_sided.recvd_pkts_tracker");
+    if (!comm->one_sided.recvd_pkts_tracker) {
+        tl_error(comm->lib, "unable to malloc for one_sided.recvd_pkts_tracker");
+        status = UCC_ERR_NO_MEMORY;
+        goto failed;
+    }
+
+    comm->one_sided.sendbuf_memkey_list = ucc_calloc
+            (1, comm->commsize * sizeof(ucc_tl_mlx5_mcast_slot_mem_info_t),
+             "one_sided.sendbuf_memkey_list");
+    if (!comm->one_sided.sendbuf_memkey_list) {
+        tl_error(comm->lib, "unable to malloc for one_sided.sendbuf_memkey_list");
+        status = UCC_ERR_NO_MEMORY;
+        goto failed;
+    }
+
+    /* below data structures are used in async design only */
+    comm->one_sided.slot_size    = comm->one_sided.reliability_scheme_msg_threshold
+                                           + ONE_SIDED_SLOTS_INFO_SIZE;
+    one_sided_total_slots_size   = comm->one_sided.slot_size *
+                                            ONE_SIDED_SLOTS_COUNT * sizeof(char);
+    comm->one_sided.slots_buffer = (char *)ucc_calloc(1, one_sided_total_slots_size,
+                                                      "one_sided.slots_buffer");
+    if (!comm->one_sided.slots_buffer) {
+        tl_error(comm->lib, "unable to malloc for one_sided.slots_buffer");
+        status = UCC_ERR_NO_MEMORY;
+        goto failed;
+    }
+    comm->one_sided.slots_mr = ibv_reg_mr(comm->ctx->pd, comm->one_sided.slots_buffer,
+                                          one_sided_total_slots_size, IBV_ACCESS_LOCAL_WRITE |
+                                          IBV_ACCESS_REMOTE_READ | IBV_ACCESS_REMOTE_WRITE);
+    if (!comm->one_sided.slots_mr) {
+        tl_error(comm->lib, "unable to register for one_sided.slots_mr");
+        status = UCC_ERR_NO_RESOURCE;
+        goto failed;
+    }
+    
+    /* this array holds local information about the slot status that was read from remote ranks */
+    comm->one_sided.remote_slot_info = ucc_calloc(1, comm->commsize * ONE_SIDED_SLOTS_INFO_SIZE,
```

Comment:
simply checking, do you intend to use this macro here, or should it be `ONE_SIDED_SLOTS_COUNT`?

Reply:
We intend to use this ````ONE_SIDED_SLOTS_INFO_SIZE```` here. This array holds the info that we read from remote processes.